xbitinfo/bitround.py

```diff
@@ -171,21 +178,42 @@ def bitround_along_dim(
     ... )
     >>> (ds - ds_bitrounded_along_lon)["air"].isel(time=0).plot()  # doctest: +ELLIPSIS
     <matplotlib.collections.QuadMesh object at ...>
+
+    # Test keepbits parameter
+    def test_bitround_along_dim_keepbits():
+    ds = xr.Dataset({'temp': (('lat', 'lon', 'time'), np.random.rand(3, 3, 3))},
+                    coords={'lat': np.arange(3), 'lon': np.arange(3), 'time': np.arange(3)})
+    info_per_bit = {'temp': np.array([0.5, 0.25, 0.125])}
+    ds_bitrounded = bitround_along_dim(ds, info_per_bit, dim='lat', keepbits=2)
+    assert ds_bitrounded.temp.dtype == 'float32'
+    assert ds_bitrounded.temp.shape == (3, 3, 3)
+    assert (ds.temp.isel(lat=slice(0, 2)) - ds_bitrounded.temp.isel(lat=slice(0, 2))).sum() == 0.0
+    assert (ds.temp.isel(lat=slice(2, None)) - ds_bitrounded.temp.isel(lat=slice(2, None))).sum() != 0.0
+
+
```

Comment:
Hi @Ishaanj18,
Thank you for adding this test. It is currently however not executed as it is only treated as comment in the docstring. The doctest syntax starts with `>>>` for commands and `` for the expected output.
As the test is more elaborate, you should move these lines to https://github.com/observingClouds/xbitinfo/blob/main/tests/test_bitround.py

Reply:
There you can also test if the `ValueError` is raised when expected.